icu4j/main/core/src/main/java/com/ibm/icu/text/RuleBasedBreakIterator.java

```diff
@@ -1229,16 +1279,20 @@ void populateDictionary(int startPos, int endPos,
                     break;
                 }
 
-                // We now have a dictionary character. Get the appropriate language object
-                // to deal with it.
-                LanguageBreakEngine lbe = getLanguageBreakEngine(c);
+                if (excludedFromDictionaryBreak(c)) {
+                    CharacterIteration.next32(fText);
+                } else {
+                    // We now have a dictionary character. Get the appropriate language object
+                    // to deal with it.
+                    LanguageBreakEngine lbe = getLanguageBreakEngine(c);
+
+                    // Ask the language object if there are any breaks. It will add them to the cache and
+                    // leave the text pointer on the other side of its range, ready to search for the next one.
+                    if (lbe != null) {
+                        foundBreakCount += lbe.findBreaks(fText, rangeStart, rangeEnd, fBreaks, fPhraseBreaking);
```

Comment:
I haven't looked in detail at this, but it appears that this wouldn't catch the case where a character before rangeEnd should be excluded.

Reply:
so... how is that behavior specified in UTS 35 + UAX 29 + UAX 14? 
Could we have a test case for that?

Reply:
Here's what I mean. The meaning of dx-xxxx is that none of the xxxx characters will be processed by the break iterators.

So say that 't' stands for Thai, and . stands for other characters.  

ttttt......ttttttttttt......ttttttt.......

With dx-thai, break iterators must only act on the dots (non-thai)

With your code, the iterator would skip over the first ttttt and start at the first non-Thai (the first dot)

However, I see nothing in the code that would prevent the iterator from continuing at least part-way into into the second group of ttttttttttt.

Reply:
So... notice this part is inside a function
populateDictionary(int startPos, int endPos,...)
If you have the text "ttttt......ttttttttttt......ttttttt......."
so the first "ttttt" is in index 0-5, and the second "ttttttttttt" is in index 11-22, 28-35
the upper call will first call this with  
populateDictionary(0, 5...)
then
populateDictionary(11, 22...)
populateDictionary(28, 35...)

and the code put in the break between 0 - 5 into fBreak first call, the breaks between 11-22 into fBreak second call and the breaks between 28-35 into fBreak the third call
and the upper caller will figure out how to break the ...

With my change, when we hit any t, if (excludedFromDictionaryBreak(c)) willl return true and therrefore just advance the iterator till 5 and return out of the loop.

Reply:
Mark, I add the following to unit test to show it does work
```
    bi->setText(UnicodeString(u"aaอออaaaaaอออ    aaaa"));
```